components/Banner/BannerWithImage.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react'
+import styled from 'styled-components'
+
+import { propTypes, defaultProps } from './propTypes'
+import { rem, headerFont } from '../../styling/typography'
+import {
+  extraExtraSmallSpacing,
+  smallSpacing,
+  mediumSpacing,
+  extraLargeSpacing,
+} from '../../styling/sizes'
+import Link from '../Link'
+import Button from '../Button'
+import Heading from './Headline'
+import { background } from './styles'
+import { lgBreakpoint, mdBreakpoint } from '../../styling/breakpoints'
+
+const Container = styled.div`
+  margin-right: 0;
+  margin-left: 0;
+`
+
+const Image = styled.div`
+  background: url(${({ image }) => image}) center no-repeat;
+  background-size: cover;
+  min-height: 300px;
+`
+
+const ContentContainer = styled.span`
+  ${background};
+  padding: ${smallSpacing};
+  color: white;
+  
+  @media(min-width: ${mdBreakpoint}) {
+    padding: ${mediumSpacing};
+  }
+  
+  @media(min-width: ${lgBreakpoint}) {
+    padding: ${mediumSpacing} ${extraLargeSpacing};
+  }
+`
+
+const SubHeading = styled.span`
+  display: block;
+  font-size: ${rem('24px')};
+  font-family: ${headerFont};
+  font-weight: 500;
+`
+
+const ExtendedHeading = Heading.extend`
+  display: block;
+  text-align: left;
+  margin-bottom: ${extraExtraSmallSpacing};
+`
+const Description = styled.span`
+  display: block;
+  margin-bottom: ${smallSpacing};
+`
+
+const BannerWithImage = ({
+  subHeadline,
+  headline,
+  description,
+  image,
+  buttonText,
+  buttonLink,
+  showImageOnRight,
+}) => {
+  const content = (
+    <ContentContainer className="col-md-8" key="content">
+      {subHeadline && <SubHeading>{subHeadline}</SubHeading>}
+      <ExtendedHeading>{headline}</ExtendedHeading>
+      {description && <Description>{description}</Description>}
+      <Link route={buttonLink} passHref>
+        <Button type="banner">{buttonText}</Button>
+      </Link>
+    </ContentContainer>
+  )
+  const img = <Image className="col-md-4" alt="" image={image} key="img" />
+
+  return (
+    <Container className="row">
+      {showImageOnRight ? [content, img] : [img, content]}
```

Comment:
I guess you wouldn't need the strange-looking `key` properties on ContentContainer and Image if you didn't return an array here. What about just doing:

```
{content}
{img}
```

?

Reply:
I tried it but it would look a little bit different than that:

Reply:
Ah, you really need that fragment thing? OK.

Reply:
I thought () was enough.

Reply:
I will keep it as it then.